packages/deployments/contracts/contracts/messaging/connectors/optimism/OptimismHubConnector.sol

```diff
@@ -61,133 +67,85 @@ contract OptimismHubConnector is HubConnector, BaseOptimism {
    */
-  function processMessageFromRoot(
-    address _target,
-    address _sender,
-    bytes memory _message,
-    uint256 _messageNonce,
-    L2MessageInclusionProof memory _proof
-  ) external {
+  function processMessageFromRoot(Types.WithdrawalTransaction memory _tx) external {
     // verify the sender is the l2 contract
-    require(_sender == mirrorConnector, "!mirrorConnector");
+    require(_tx.sender == PredeployAddresses.L2_CROSS_DOMAIN_MESSENGER, "!l2sender");
 
     // verify the target is this contract
-    require(_target == address(this), "!this");
-
-    // Get the encoded data
-    bytes memory xDomainData = _encodeXDomainCalldata(_target, _sender, _message, _messageNonce);
-
-    require(_verifyXDomainMessage(xDomainData, _proof), "!proof");
-
-    // NOTE: optimism seems to pad the calldata sent in to include more than the expected
-    // 36 bytes, i.e. in this transaction:
-    // https://blockscout.com/optimism/goerli/tx/0x440fda036d28eb547394a8689af90c5342a00a8ca2ab5117f2b85f54d1416ddd/logs
-    // the corresponding _message is:
-    // 0x4ff746f60000000000000000000000000000000000000000000000000000000000000020000000000000000000000000000000000000000000000000000000000000002027ae5ba08d7291c96c8cbddcc148bf48a6d68c7974b94356f53754ef6171d757
-    //
-    // this means the length check and byte parsing used in the `ArbitrumHubConnector` would
-    // not work here. Instead, take the back 32 bytes of the string
-
-    // NOTE: TypedMemView only loads 32-byte chunks onto stack, which is fine in this case
-    bytes29 _view = _message.ref(0);
-    bytes32 root = _view.index(_view.len() - 32, 32);
-
-    if (!processed[root]) {
-      // set root to processed
-      processed[root] = true;
-      // update the root on the root manager
-      IRootManager(ROOT_MANAGER).aggregate(MIRROR_DOMAIN, root);
-
-      emit MessageProcessed(abi.encode(root), msg.sender);
-    } // otherwise root was already sent to root manager
+    require(_tx.target == address(this), "!this");
+
+    require(_verifyXDomainMessage(_tx), "!proof");
+
+    // Trigger the call to the target contract. We use a custom low level method
+    // SafeCall.callWithMinGas to ensure two key properties
+    //   1. Target contracts cannot force this call to run out of gas by returning a very large
+    //      amount of data (and this is OK because we don't care about the returndata here).
+    //   2. The amount of gas provided to the call to the target contract is at least the gas
+    //      limit specified by the user. If there is not enough gas in the callframe to
+    //      accomplish this, `callWithMinGas` will revert.
+    // Additionally, if there is not enough gas remaining to complete the execution after the
+    // call returns, this function will revert.
+    bool success = SafeCall.callWithMinGas(_tx.target, _tx.gasLimit, _tx.value, _tx.data);
```

Comment:
this is just going to `.call` this contract, we should skip the actual call here and rely on the call to the `RootManager` in `_processMessage`